src/osp/Active/scene.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2020 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "../types.h"
+#include "activetypes.h"
+
+#include <entt/entity/entity.hpp>
+
+namespace osp::active
+{
+
+/**
+ * Component for transformation (in meters)
+ */
+struct ACompTransform
+{
+    osp::Matrix4 m_transform;
+};
+
+struct ACompTransformControlled { };
+
+struct ACompTransformMutable{ bool m_dirty{false}; };
+
+/**
+ * Added to an entity to mark it for deletion
+ */
+struct ACompDelete{ };
+
+struct ACompName
+{
+    ACompName(std::string name) : m_name(name) { }
+    std::string m_name;
```

Comment:
This would be a good place for the shared_string class. Write-once, read-many.

Reply:
I've considered this, but it doesn't compile on my system at least. You mentioned on the Discord that it might be an outdated version of the file. Would it be a good idea to fix this in this PR?

Reply:
hrmm, could you remind me what the compile errors you ran into were?

maybe i need to make a unit test folder for osp-magnum so we can start adding tests.

Reply:
but if you want to fix it, please feel free. I don't know that i have a different copy really, it's been long enough that it might have gotten deleted from my local machine.

Reply:
I'm not sure how to fix it; I put my attempt on [this branch.](https://github.com/Capital-Asterisk/osp-magnum/tree/hierarchy-improvements2-sharedstring)